grpc/errors.go

```diff
@@ -66,50 +67,60 @@ func unwrapError(err error, md metadata.MD) error {
 	if err == nil {
 		return nil
 	}
-	if errTypeStrs, ok := md["errortype"]; ok {
-		unwrappedErr := grpc.ErrorDesc(err)
-		if len(errTypeStrs) != 1 {
-			return berrors.InternalServerError(
-				"multiple errorType metadata, wrapped error %q",
-				unwrappedErr,
-			)
-		}
-		errType, decErr := strconv.Atoi(errTypeStrs[0])
-		if decErr != nil {
-			return berrors.InternalServerError(
-				"failed to decode error type, decoding error %q, wrapped error %q",
-				decErr,
-				unwrappedErr,
-			)
-		}
-		outErr := berrors.New(berrors.ErrorType(errType), unwrappedErr)
-		if subErrsJSON, ok := md["suberrors"]; ok {
-			if len(subErrsJSON) != 1 {
-				return berrors.InternalServerError(
-					"multiple suberrors metadata, wrapped error %q",
-					unwrappedErr,
-				)
-			}
-			var suberrs []berrors.SubBoulderError
-			if err := json.Unmarshal([]byte(subErrsJSON[0]), &suberrs); err != nil {
-				return berrors.InternalServerError(
-					"error unmarshaling suberrs JSON %q, wrapped error %q",
-					subErrsJSON[0],
-					unwrappedErr,
-				)
-			}
-			var berr *berrors.BoulderError
-			if errors.As(outErr, &berr) {
-				outErr = berr.WithSubErrors(suberrs)
-			} else {
-				return fmt.Errorf(
-					"expected type of outErr to be %T got %T: %q",
-					berr, outErr,
-					outErr.Error(),
-				)
-			}
-		}
+
+	unwrappedErr := status.Convert(err).Message()
```

Comment:
Reference: https://pkg.go.dev/google.golang.org/grpc#ErrorDesc

> ErrorDesc returns the error description of err if it was produced by the rpc system. Otherwise, it returns err.Error() or empty string when err is nil.
> Deprecated: use status.Convert and Message method instead.